tests/server_encode.rs

```diff
@@ -94,16 +94,16 @@ date: {DATE}
             Method::Get,
             res,
             r#"HTTP/1.1 200 OK
-transfer-encoding: chunked
-date: {DATE}
 content-type: application/octet-stream
+date: {DATE}
+transfer-encoding: chunked
 
-1
-h
 5
-ello 
+hello
 5
-world
+ worl
+1
+d
```

Comment:
why does the chunked encoding end up like this in particular? Surely the buffered chunk size is large enough to fit `"Hello World"`?

Reply:
in this case i'm intentionally exercising the chunking by passing in 10 as the first arg to assert_encoded. with a buffer of 10, we get `\r\n5\r\nhello` for a total of 10 bytes. if we specified 12, we'd get 7 bytes of data, etc

Reply:
(and there's some math in there to figure out the maximum number of hex digits required for the maximum buffer length, so we never ask for more data from the body than we can fit into the buffer along with the framing)

Reply:
and—to be clear—10 is an extremely unrealistic number. it just was convenient for testing

Reply:
ah I missed the `10` argument